### ISSUE TITLE
Text wrapping splits words containing non-ASCII characters

### ISSUE THREAD

**Issue by MartinZikmund** [verybadcat/CSharpMath#135]
### Describe the bug

Text wrapping does not properly work for accented characters like ěščřžýáíé.

### To Reproduce

1. Open sample app for Xamarin.Forms
2. Select **Text** tab
3. Enter: ščěěščž čšřžščřž  čšřžščř  čšřžščř čšřžščř  čšřžščř  čšřžščř
4. Resize the window. Notice the line breaks anywhere within the words

### Expected behavior

Words with accented characters should not be broken between lines.

### Environment (please complete the following information):
 - Platform: CSharpMath.SkiaSharp (in general, not just XF)
 - Package Version or Commit: latest
 - Device: Windows

## Maintainer triage (2026-08-30)

This issue now tracks incorrect word breaking for **non-ASCII words**, including the Czech reproduction above and the Russian reproduction from #192. Font loading, missing glyphs, bold/italic faces, and non-seekable streams are separate concerns and are out of scope here.

### Acceptance criteria

- Text wrapping does not split Czech or Russian words at arbitrary character boundaries.
- Break opportunities remain correct around spaces and punctuation.
- Regression tests cover representative Czech and Cyrillic input through the Typography-backed text parser/typesetter.

**Comment by Happypig375** [comment]
@prepare Probably a bug of Typography.TextBreak.

**Comment by Happypig375** [comment]
Confirmed a bug of Typography.TextBreak.

**Comment by prepare** [comment]
This Typography bug was fixed, 

see => https://github.com/LayoutFarm/Typography/commit/f350e2268ab2236d07d58a49647cf1e60a6764c4


![include_extenedLatin](https://user-images.githubusercontent.com/7447159/85256737-e0e9dd00-b48e-11ea-866e-186c65834667.png)

**Comment by MartinZikmund** [comment]
@prepare Thanks a lot!

**Comment by Happypig375** [comment]
Still need to fix this for diacritics via LaTeX commands instead of Unicode, i.e. `a\r aa` should be one group.

**Comment by BenjaNapo** [comment]
![Screenshot_1593169397](https://user-images.githubusercontent.com/25958425/85850700-9c6f8180-b7ad-11ea-97b7-a38a653d9ba8.png)
That's what happen when I try to show chinese text, is this relate to this issue?

**Comment by Happypig375** [comment]
You should provide your own font like in #142.

**Comment by Happypig375** [comment]
@BenjaNapo See https://github.com/verybadcat/CSharpMath/issues/20.

**Comment by BenjaNapo** [comment]
> @BenjaNapo See #20.

I wrote in the constructor of App.cs:
`CSharpMath.Settings.GlobalTypefaces = new Typography.OpenFont.OpenFontReader().Read(File.Open("Cabin-Regular.ttf", FileMode.Open));`

but it says that GlobalTypefaces is an onlyread attribute

**Comment by Happypig375** [comment]
For GlobalTypefaces, you call `Add`.

**Comment by BenjaNapo** [comment]
> For GlobalTypefaces, you call `Add`.

### App.xaml
 ```
            <OnPlatform x:Key="Regular" x:TypeArguments="x:String">
                <On Platform="Android" Value="Cabin-Regular.ttf#Cabin" />
                <On Platform="iOS" Value="Cabin-Regular" />
            </OnPlatform>
```
### App.cs
 ```
        public App()
        {
            InitializeComponent();
            [... various initialization ...]
            object s = new object();
            Resources.TryGetValue("FontAwesomeRegular", out s);
            CSharpMath.Settings.GlobalTypefaces.Add(new Typography.OpenFont.OpenFontReader().Read(s as Stream));
        }
```

Sorry but I never imported Font from code, where did I go wrong?

**Comment by Happypig375** [comment]
Are you trying to override the font? The default is `AddSupplement`. For overrides, use `AddOverride`.

**Comment by BenjaNapo** [comment]
> Are you trying to override the font? The default is `AddSupplement`. For overrides, use `AddOverride`.

Changed with:
```
            object s = new object();
            Resources.TryGetValue("Regular", out s);
            Typeface t = new Typography.OpenFont.OpenFontReader().Read(s as Stream);
            CSharpMath.Settings.GlobalTypefaces.AddOverride(t);
```
But 3rd line throws ArgumentNullException, I think that the problem is in reading font file in this way

**Comment by Happypig375** [comment]
The null means that `s` is probably not a `Stream`. Please check.

**Comment by BenjaNapo** [comment]
```
        public async void LoadFont() {
            using (var stream = await FileSystem.OpenAppPackageFileAsync("Helvetica-Bold.otf"))
            {
                Typeface t = new Typography.OpenFont.OpenFontReader().Read(stream as Stream);
                CSharpMath.Settings.GlobalTypefaces.AddOverride(t);
            }
        }
```
Now it loads the font from the stream but crash on the 4th line (Typeface t = ...) throwing this exception:
`System.NotSupportedException: 'Specified method is not supported.'`

**Comment by Happypig375** [comment]
Try following https://github.com/verybadcat/CSharpMath#sourcelink-for-ci-packages and debug into the source of Typography.

**Comment by Happypig375** [comment]
@BenjaNapo Or you can upload a zipped file containing your file and let me investigate what changes are needed from Typography's side.

**Comment by Happypig375** [comment]
This bug's fix depends on a decision on updating Typography in LayoutFarm/Typography#192.
The fix for 
>Still need to fix this for diacritics via LaTeX commands instead of Unicode, i.e. `a\r aa` should be one group.

is scheduled after #143 and #58.

**Comment by BenjaNapo** [comment]
https://drive.google.com/file/d/1zlvyQ2XXvb0fQJ8k-otYZ4hiR_8L3H-d/view?usp=sharing

Here you are, with this you could note all the issues that I had 'til now (text alignment, change font and height of textview)

Thanks for all the help ^^ Hope this will contribute with the evolution of the library

**Comment by Happypig375** [comment]
@BenjaNapo #144 should fix "text alignment" and "height of textview".

This leaves "change font" as to-do.

**Comment by Happypig375** [comment]
@BenjaNapo I can't download the file. 
![image](https://user-images.githubusercontent.com/19922066/87854568-62754380-c945-11ea-8b41-a1ee2d106b52.png)

I have requested access.

**Comment by BenjaNapo** [comment]
> I have requested access.

Ow, sorry, thought that just the link will work. Does it work now?

**Comment by Happypig375** [comment]
Works now, thanks

**Comment by Happypig375** [comment]
@BenjaNapo I'm getting an exception other than the one you wrote.
![image](https://user-images.githubusercontent.com/19922066/87867269-d5ba9c00-c9bd-11ea-89f0-8db52e02a318.png)

**Comment by Happypig375** [comment]
@BenjaNapo I'm not getting the `NotSupportedException`.
![image](https://user-images.githubusercontent.com/19922066/87867325-70b37600-c9be-11ea-9161-bea073ac5985.png)

**Comment by BenjaNapo** [comment]
Still not working, tried as you to put the method inside async OnStart but nothing
![2020-07-23](https://user-images.githubusercontent.com/25958425/88281793-91ded400-cce8-11ea-8878-0af8a57df117.png)
It crash at line 53, and, as you can see, stream.Length and stream.Position are not setted, so probably the Read try to use them

**Comment by Happypig375** [comment]
What is the stack trace?

**Comment by BenjaNapo** [comment]
Do you mean this?
![2020-07-23 (2)](https://user-images.githubusercontent.com/25958425/88306161-adf36d00-cd0a-11ea-8d8f-aef5a1dd337e.png)

**Comment by Happypig375** [comment]
I mean the stack trace of the exception.

**Comment by BenjaNapo** [comment]
Ehm... How can I ?

**Comment by Happypig375** [comment]
![image](https://user-images.githubusercontent.com/19922066/88359798-936ed180-cda6-11ea-8d93-4af6b3ba4d8c.png)

**Comment by BenjaNapo** [comment]
As you can see I don't have the "View Details" option 🙄 Probably is because I'm working on Android device and not using UWP

![2020-07-24 (1)](https://user-images.githubusercontent.com/25958425/88378090-96ff5a00-cda0-11ea-8ddd-dc32805334e5.png)

**Comment by MartinZikmund** [comment]
@Happypig375 Should this be working in the latest preview NuGet package (0.5.0-alpha4) or is it not yet released?

**Comment by Happypig375** [comment]
Yes, the fix should be included in 0.5.0-alpha4.

**Comment by MartinZikmund** [comment]
It seems it does not work for me 🤔 :
![image](https://user-images.githubusercontent.com/1075116/92991535-cac74800-f4e4-11ea-8728-55581205f5c7.png)

For example on the last line of the text is the work "světovou" broken as "svě" and "tovou". Was the Typography module bumped to the updated version?

**Comment by Happypig375** [comment]
Ohhh. Yea I misrecognized this issue as the ones that were fixed. Yes this depends on updating Typography currently, but I am looking to refactor to reuse the math parser for text, so the Typography.TextBreak dependency will be gone by then. This is targeted at 0.5.0-beta.

**Comment by BenjaNapo** [comment]
In the end I resolved in this way:
```
            Stream stream = await FileSystem.OpenAppPackageFileAsync("GTEesti-Light.ttf");
            MemoryStream ms = new MemoryStream();
            stream.CopyTo(ms);
            ms.Position = 0;
            OpenFontReader ofr = new OpenFontReader();
            Typeface t = ofr.Read(ms);
            CSharpMath.Settings.GlobalTypefaces.AddOverride(t);
```

**Comment by Happypig375** [comment]
Oh... Maybe the stream returned by `FileSystem.OpenAppPackageFileAsync` is not seekable.

**Comment by BenjaNapo** [comment]
Yep, indeed the NotSupportedException was in Length/Position attributes 😅 I think that now we can close the issue (?)

**Comment by BenjaNapo** [comment]
> ![Screenshot_1593169397](https://user-images.githubusercontent.com/25958425/85850700-9c6f8180-b7ad-11ea-97b7-a38a653d9ba8.png)
> That's what happen when I try to show chinese text (appear squares), is this relate to this issue?

Anyway I'm still having this issue, on Label chinese text is showed (so it's a not problem of the font)...

**Comment by MartinZikmund** [comment]
Hey, I have updated to latest package (beta), but the problem seems to persist:
![image](https://user-images.githubusercontent.com/1075116/98253459-586e6380-1f7b-11eb-8f1d-3e3833582069.png)

**Comment by BenjaNapo** [comment]
> Hey, I have updated to latest package (beta), but the problem seems to persist:
> ![image](https://user-images.githubusercontent.com/1075116/98253459-586e6380-1f7b-11eb-8f1d-3e3833582069.png)
> 
> Also noticed the accented characters don't play well with bold/italics, is that a known issue?

Try to load an italic font

**Comment by charlesroddie** [comment]
> Also noticed the accented characters don't play well with bold/italics, is that a known issue?

This is tracked here: https://github.com/verybadcat/CSharpMath/issues/142

However it will be a few weeks before I can work on this.

**Comment by MartinZikmund** [comment]
Hi @Happypig375, regarding the line break fix for words with accented characters, is there an estimate when a preview package with this could be released please :-) ?

**Comment by Happypig375** [comment]
@MartinZikmund I don't have the capacity to put in work in the near future. It's really best to just fork and submit a PR for now.

**Comment by MartinZikmund** [comment]
@Happypig375 of course, understood 👍. Just to check - is there something missing on the text wrapping code-wise or is it just about releasing a new package? I thought @prepare adjusted the line breaks in Typography, which should transitively fix the problem for CSharpMath, if we update to that version.

**Comment by Happypig375** [comment]
See https://github.com/verybadcat/CSharpMath/issues/107#issuecomment-650762446.

**Comment by charlesroddie** [comment]
@MartinZikmund I said I would do the font work but am not using TextPainter any more. I don't think I have time to progress with this by myself in the very near future but can collaborate with you on a PR. Weekends best.

**Comment by Happypig375** [comment]
Triage update: I narrowed this to the still-actionable non-ASCII word-breaking bug and added acceptance criteria. #192 is the same underlying Cyrillic case and will be consolidated here; the unrelated font-loading discussion remains preserved in the history but is no longer part of this issue's scope.
